src/main/webapp/app/js/goal-page.js

```diff
@@ -135,4 +143,22 @@ function saveEditGoal() {
             return false;
         }
     });
+}
+
+function setStatusAchieved(id) {
+    fetch(path + "/api/goal/mygoals/" + id, {
+        "method": "POST",
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+    }).then(function (response) {
+        if (response.status === 204) {
+            onLoad();
+        } else {
+            console.log(response);
```

Comment:
Might be worth it to extract `console.log()` and `alert()` to a single function which reports the error, and reuse that function over the code.

Reply:
alert() extracted to function displayErrors() and removed all console.log()